packages/@sanity/cli/src/util/journeyConfig.ts

```diff
@@ -0,0 +1,270 @@
+import fs from 'fs/promises'
+import path from 'path'
+import {format} from 'prettier'
+import {Worker} from 'worker_threads'
+
+import {
+  type ArrayDefinition,
+  type BaseSchemaDefinition,
+  type DocumentDefinition,
+  type ObjectDefinition,
+} from '../../../types'
+import {type CliApiClient} from '../types'
+import {getCliWorkerPath} from './cliWorker'
+
+/**
+ * A Journey schema is a server schema that is saved in the Journey API
+ */
+
+type JourneySchemaWorkerData = {
+  schemasPath: string
+  useTypeScript: boolean
+  projectId: string
+}
+
+type JourneySchemaWorkerResult = {type: 'success'} | {type: 'error'; error: Error}
+
+type JourneyConfigResponse = {
+  projectId: string
+  datasetName: string
+  displayName: string
+  isFirstProject: boolean // Always true for now, making it compatible with the existing getOrCreateProject
+}
+
+type DocumentOrObject = DocumentDefinition | ObjectDefinition
+type SchemaObject = BaseSchemaDefinition & {type: string; fields?: SchemaObject[]}
+
+/**
+ * Fetch a Journey schema from the Sanity schema club API and write it to disk
+ */
+export async function getAndWriteJourneySchema(data: JourneySchemaWorkerData): Promise<void> {
+  const {schemasPath, useTypeScript, projectId} = data
+  try {
+    const documents = await fetchJourneySchema(projectId)
+    const fileExtension = useTypeScript ? 'ts' : 'js'
+
+    // Write a file for each schema
+    for (const document of documents) {
+      const filePath = path.join(schemasPath, `${document.name}.${fileExtension}`)
+      await fs.writeFile(filePath, await JourneySchemaToFileContents(document))
+    }
+    // Write an index file that exports all the schemas
+    const indexContent = assembeJourneyIndexContent(documents)
+    await fs.writeFile(path.join(schemasPath, `index.${fileExtension}`), indexContent)
+  } catch (error) {
+    throw new Error(`Failed to fetch remote schema: ${error.message}`)
+  }
+}
+
+/**
+ * Run the getAndWriteJourneySchema in a worker thread
+ *
+ * @param workerData - The worker data to pass to the worker thread
+ */
+export async function getAndWriteJourneySchemaWorker(
+  workerData: JourneySchemaWorkerData,
+): Promise<void> {
+  const workerPath = await getCliWorkerPath('getAndWriteJourneySchema')
+  return new Promise((resolve, reject) => {
+    const worker = new Worker(workerPath, {
+      workerData,
+      env: {
+        // eslint-disable-next-line no-process-env
+        ...process.env,
+        // Dynamic HTTPS imports are currently behind a Node flag
+        NODE_OPTIONS: '--experimental-network-imports',
+        NODE_NO_WARNINGS: '1',
+      },
+    })
+    worker.on('message', (message: JourneySchemaWorkerResult) => {
+      if (message.type === 'success') {
+        resolve()
+      } else {
+        reject(message.error)
+      }
+    })
+    worker.on('error', reject)
```

Comment:
Might make sense to provide context about the error here as well, so instead of rejecting directly I suggest adding context to the error message, e.g.:

```suggestion
worker.on('error', error => {
  error.message = `Import schema worker failed: ${error.message}`
  reject(error)
})
```